### DISCUSSION TITLE
How to get into correct directory when running Rstudio on Mox?

### DISCUSSION THREAD

**Discussion by aspencoyle** [RobertsLab/resources#1192] [Q&A] [answered]
Alright! So I got Rstudio to run on Mox (thanks to all the details Yaamini and Sam went into in [these] (https://github.com/RobertsLab/resources/discussions/1180) [threads](https://github.com/RobertsLab/resources/discussions/1190).

However, I'm really not sure where I actually am within Mox - how do I navigate to a folder with all my data/scripts/etc? Or really, how do I navigate anywhere at all?

Also, would it be alright to just clone my whole github repo to somewhere within /gscratch/srlab/afcoyle, or is that using up unnecessary memory?

Here's an image of where I'm at right now within RStudio:
![image](https://user-images.githubusercontent.com/71358918/116478004-4c7ca380-a832-11eb-9082-59a3f69e712e.png)

**Comment by kubu4** [comment, marked as answer]
You want the function `setwd()`. Use quotes around path. You will be able to use autocomplete to help fill in the full path, if you need/want.

Alternatively, and best practice, create a new R Project. You'll be able to "browse" to your desired location.

Finally, yes, you can clone repos if you want; we should have sufficient storage.

**Reply by aspencoyle** [reply]
Oh man, duh - thanks for fixing that!

And great to know - I'll clone that repo over!

**Reply by kubu4** [reply]
Also, reminder, RStudio Server does not have internet access. So, with git repos, you'll be able to commit changes via RStudio, but you won't be able to push changes. You'll have to do this via login node, using the guy command line functions.

**Comment by kubu4** [comment]
Side note, I discovered today that graphics won't display in R Markdown. I need to install a system library. I'll tackle that tomorrow and get a new Singularity container transferred to Mix.

**Reply by aspencoyle** [reply]
Gotcha! As an update, I'm getting graphics to display alright in .Rmd files (although haven't tried knitting them yet - assuming that's probably what you mean)

Finally, question with R projects - I can't see my R project (or any files from my current WD) within the files tab). Is there a way to open an existing R project from the console?

I suppose I can always create a new one, but it'd likely be easier to just use the one that's part of my repo already..

**Reply by kubu4** [reply]
> Is there a way to open an existing R project from the console?

No need to use the console. Can use the RStudio interface:

1. File > Open Project

2. Click `...` on right side of dialog box.

3. Paste full path to folder containing desired R Project. (e.g. `/gscratch/scrubbed/samwhite/outputs/rstudio_server_test/code/r_projects/sam/20200430_cbai_deg_multiple_conditions`)
